extensions/HuggingFaceTransformers/python/src/aiconfig_extension_hugging_face_transformers/text_generation.py

```diff
@@ -224,27 +224,31 @@ async def run_inference(
         """
         completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
         completion_data["text_inputs"] = completion_data.pop("prompt", None)
+        
+        model_name : str = aiconfig.get_model_name(prompt)
+        if isinstance(model_name, str) and model_name not in self.generators:
+            self.generators[model_name] = pipeline('text-generation', model=model_name)
+        generator = self.generators[model_name]
```

Comment:
I'm not sure how I feel about this.  Two things here. Not blocking but something to think about and if this could be done differently

1. This change means the pipeline gets loaded at execution/run time
2.  If the user has many models, all of them get tied down to this one model parser, as opposed to one per model

Reply:
Create a GH "issue" for us to keep track, good design flagging: https://github.com/lastmile-ai/aiconfig/issues/415